src/test/regress/expected/distributed_transactions.out

```diff
@@ -0,0 +1,1233 @@
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+ERROR:  cannot use 2PC in transactions involving multiple servers
+CONTEXT:  while executing command on localhost:xxxxx
+SELECT 'Transaction with duplicate error in the remote function' Testing;
+                         testing
+---------------------------------------------------------------------
+ Transaction with duplicate error in the remote function
+(1 row)
+
+BEGIN;
+INSERT INTO distributed_transaction_function.test_txn_dist VALUES (10);
+NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(xx, xx, 'xxxxxxx');
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing INSERT INTO distributed_transaction_function.test_txn_dist_900004 (intcol) VALUES (10)
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+-- This call will fail with duplicate error on the remote worker
+SELECT insert_data(11, 11);
+DEBUG:  pushing down function call in a multi-statement transaction
+DEBUG:  pushing down the function call
+NOTICE:  issuing SELECT distributed_transaction_function.insert_data(11, 11) AS insert_data
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+ERROR:  duplicate key value violates unique constraint "test_txn_dist_pkey_900030"
+DETAIL:  Key (intcol)=(11) already exists.
+CONTEXT:  SQL statement "INSERT INTO distributed_transaction_function.test_txn_dist VALUES (b)"
+PL/pgSQL function distributed_transaction_function.insert_data(integer,integer) line XX at SQL statement
+while executing command on localhost:xxxxx
+INSERT INTO distributed_transaction_function.test_txn_dist VALUES (12);
+ERROR:  current transaction is aborted, commands ignored until end of transaction block
+COMMIT;
+SELECT 'Transaction with duplicate error in the local statement' Testing;
+                         testing
+---------------------------------------------------------------------
+ Transaction with duplicate error in the local statement
+(1 row)
+
+BEGIN;
+INSERT INTO distributed_transaction_function.test_txn_dist VALUES (20);
+NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(xx, xx, 'xxxxxxx');
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing INSERT INTO distributed_transaction_function.test_txn_dist_900000 (intcol) VALUES (20)
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+-- This call will insert both the rows locally on the remote worker
+SELECT insert_data(21, 22);
+DEBUG:  pushing down function call in a multi-statement transaction
+DEBUG:  pushing down the function call
+NOTICE:  issuing BEGIN TRANSACTION ISOLATION LEVEL READ COMMITTED;SELECT assign_distributed_transaction_id(xx, xx, 'xxxxxxx');
+DETAIL:  on server postgres@localhost:xxxxx connectionId: xxxxxxx
+NOTICE:  issuing SELECT distributed_transaction_function.insert_data(21, 22) AS insert_data
```

Comment:
shouldn't this give a "queries must filter by distribution argument" error?

Reply:
I guess because the worker does not have the pg_dist_object metadata yet

Reply:
That's correct, as the metadata is not synced (yet) on the worker we are not exercising that piece of code. On a related note, I made an unsuccessful attempt to merge the changes.